packages/core/src/Synchronizer.ts

```diff
@@ -742,6 +777,23 @@ export class Synchronizer extends EventEmitter {
             timestamp,
             blockNumber: event.blockNumber,
         })
+
+        db.upsert('Epoch', {
```

Comment:
I'd rather not put the `attest` flag on the epoch object. Upserting is wonky with `Epoch` objects because they have a compound index. The anondb implementation needs a single unique key to determine if an existing document exists. This is why in the `handleEpochEnded` function i use a find, then a create or update call after.

Instead of storing `attest: boolean` on the epoch we can do 

```js
db.findOne('Attestation', { where: { attesterId, epoch }})
```

Or we can keep the `attest` field on the `Epoch` if this upsert is changed to a `find` and then `update` or `create`.

Reply:
good idea